### PR TITLE
Add request strategy to sync teams and their members during slow sync

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* Add `TeamSyncRequestStrategy` to sync teams and their members during the new sync phase `.fetchingTeams`.

### CODE REVIEW

File: Source/Synchronization/Transcoders/TeamSyncRequestStrategy.swift [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    public var contextChangeTrackers: [ZMContextChangeTracker] {
+        return []
+    }
+
+    public var requestGenerators: [ZMRequestGenerator] {
+        return [teamListSync, memberSync]
+    }
+
+    fileprivate func finishSyncIfCompleted() {
+        // TODO: Check if all teams (and members?) have been fetched here
```
Comment (typfel) [inline Source/Synchronization/Transcoders/TeamSyncRequestStrategy.swift]: Can be removed?

File: Source/Synchronization/Transcoders/TeamSyncRequestStrategy.swift [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+            let members = membersPayload?.flatMap { (payload) -> Member? in
+                Team.fetchOrCreate(with: identifier, create: true, in: managedObjectContext).flatMap {
+                    Member.createOrUpdate(with: payload, in: $0, context: managedObjectContext)
+                }
+            }
+
+            // We set the flag to update a user to false as we will
+            // fetch them later in the `.fetchingUsers` sync phase
+            members?.forEach {
```
Comment (typfel) [inline Source/Synchronization/Transcoders/TeamSyncRequestStrategy.swift]: Is this necessary? I assume the flag would be reset when they are fetched in the `.fetchingUsers` phase

File: Tests/Source/Synchronization/Transcoders/TeamSyncRequestStrategyTests.swift [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+
+            // when
+            mockSyncStatus.mockPhase = phase
+
+            // then
+            XCTAssertNil(sut.nextRequest(), "Should'nt generate a request in sync phase: \(phase)")
+        }
+    }
+
+    func testTahtItDoesntCreateRequestsInUnauthenticatedState() {
```
Comment (typfel) [inline Tests/Source/Synchronization/Transcoders/TeamSyncRequestStrategyTests.swift]: typo: `Taht` > `That`

File: Source/Synchronization/Transcoders/TeamSyncRequestStrategy.swift (hunk 2 above)
Comment (daehn) [inline Source/Synchronization/Transcoders/TeamSyncRequestStrategy.swift]: Good point, I will check.

File: Tests/Source/Synchronization/Transcoders/TeamSyncRequestStrategyTests.swift [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+        mockSyncStatus.mockPhase = .fetchingTeams
+
+        // when
+        mockApplicationStatus.mockSynchronizationState = .unauthenticated
+
+        // then
+        XCTAssertNil(sut.nextRequest())
+    }
+
+    func testTahtItDoesntCreateRequestsInEventProcessingState() {
```
Comment (typfel) [inline Tests/Source/Synchronization/Transcoders/TeamSyncRequestStrategyTests.swift]: typo: `Taht` > `That`

File: Source/Synchronization/Transcoders/TeamSyncRequestStrategy.swift (hunk 2 above)
Comment (daehn) [inline Source/Synchronization/Transcoders/TeamSyncRequestStrategy.swift]: It's already reset, as there shouldn't be any transcoder acting on this until the slow sync is done I think we can remove it.

**Comment by typfel** [comment]
@daehn cCan we merge this now?

**Comment by daehn** [comment]
@typfel Yes can be merged, I am working on another integration test which is still failing, but will create a separate PR for that one.

**Pull request merged**